custom/icds_reports/static/js/angular-services/date-helper.service.js

```diff
@@ -0,0 +1,19 @@
+window.angular.module('icdsApp').factory('dateHelperService', ['$location', function ($location) {
```

Comment:
This is a pattern that shows up a few times in the PR. Basically extracting code that was either already in multiple places, or about to be in multiple places and pulling it into this service so it could be shared.

The other instances of this type of change live in the own commits.